### PR TITLE
Enable analysers

### PR DESCRIPTION
**Pull request by martincostello**
## Changes

Enable `<AnalysisLevel>latest-All</AnalysisLevel>` as suggested in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/900#issuecomment-1387222338.

Within the production shipping packages there are two changes that change the public API surface. As both packages are still pre-release on NuGet.org I assumed this was fine. These changes can be reverted and suppressed if preferred.

For the test projects I erred on the side of disabling rules rather than fixing them to reduce churn.

I'm leaving this as a draft for now as it touches a lot of files so might ping a lot of different reviewers. I can also split the commits up differently and/or create smaller more targeted PRs if that's preferred.

Let me know if any of these changes warrant a note in the CHANGELOG.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #

### CODE REVIEW

File: build/Common.props [hunk 1]
```diff
@@ -7,6 +7,7 @@
     <DefineConstants>$(DefineConstants);SIGNED</DefineConstants>
     <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
     <NetFrameworkMinimumSupportedVersion>net462</NetFrameworkMinimumSupportedVersion>
+    <AnalysisLevel>latest-All</AnalysisLevel>
```
Comment (Oberon00) [inline build/Common.props:10]: latest-All sound a bit too much, can we use a fixed version here?

File: src/OpenTelemetry.Contrib.Extensions.AWSXRay/Resources/AWSEKSResourceDetector.cs [hunk 2]
```diff
@@ -41,19 +41,19 @@ public class AWSEKSResourceDetector : IResourceDetector
     public IEnumerable<KeyValuePair<string, object>> Detect()
     {
         var credentials = this.GetEKSCredentials(AWSEKSCredentialPath);
-        var httpClientHandler = Handler.Create(AWSEKSCertificatePath);
+        using var httpClientHandler = Handler.Create(AWSEKSCertificatePath);
```
Comment (Oberon00) [inline src/OpenTelemetry.Contrib.Extensions.AWSXRay/Resources/AWSEKSResourceDetector.cs:44]: Nice catch

**Comment by Oberon00** [comment]
All in all, this PR seems a bit too large. The nontrivial changes should probably go to separate PRs (separately dis/enabled before)

File: build/Common.props (hunk 1 above)
Comment (Kielek) [reply build/Common.props:10]: There is an agreement to use this approach on SDK repository. I think we could do the same here. See https://github.com/open-telemetry/opentelemetry-dotnet/issues/3958

**Comment by Kielek** [comment]
> All in all, this PR seems a bit too large. The nontrivial changes should probably go to separate PRs (separately dis/enabled before)

I think that enabling this feature project by project (including tests) could be easier to review. If you agree, it will be great to create issue to link all changes, similar to #894.

@TimothyMothra, fyi as you working on similar issue in SDK repository.

**Comment by Oberon00** [comment]
I was thinking of doing the trivial warnings (such as adding static) in one PR, and for the more involved changes, doing it per warning, or per project, however it clusters better.

File: build/Common.props (hunk 1 above)
Comment (Oberon00) [reply build/Common.props:10]: Makes sense.

**Comment by martincostello** [comment]
As noted in the description, I'm happy to chunk this up into N commits/PRs (whatever works best for you), but doing _all_ the changes and pushing up a draft as a starting point was the easiest way for me to do the majority of the work in the first instance. The easiest way to fix with it turned on was per-file, per-project.

**Comment by martincostello** [comment]
Have opened #4144 to track doing these changes in a series of separate pull requests.

**Pull request closed without merging**